tests/test/constants.rs

```diff
@@ -105,6 +122,99 @@ fn generic_impl() {
     }
 }
 
+#[test]
+fn unevaluated_impl() {
+    test! {
+        program {
+            struct S<const N> {}
+
+            trait Trait {}
+
+            impl Trait for S<3?> {}
+        }
+
+        goal {
+            exists<const N> {
+                S<N>: Trait
+            }
+        } yields {
+            "Unique; substitution [?0 := 3?], lifetime constraints []"
```

Comment:
Ah, interesting. I have to think if this is something we want to be able to support. (Giving solutions with unevaluated consts). It's pretty clear that these would *have* to be evaluable, since they would have to pass WF/coherence checks.